src/libostree/ostree-impl-system-generator.c

```diff
@@ -163,6 +164,32 @@ _ostree_impl_system_generator (const char *ostree_cmdline,
   if (found_var_mnt)
     return TRUE;
 
+  struct statvfs stvfsbuf;
+  if (statvfs ("/sysroot", &stvfsbuf) == -1)
+    return glnx_throw_errno_prefix (error, "statvfs(/sysroot)");
+  const gboolean sysroot_currently_writable = ((stvfsbuf.f_flag & ST_RDONLY) == 0);
+  const char *ordering_value;
+  /* For ostree as originally envisioned, e.g. Fedora Atomic Host, the system starts
+   * with the rootfs mounted ro (kernel default), and then gets remounted by
+   * systemd.
+   * 
+   * However, Fedora CoreOS is an Ignition based system and starts out writable
+   * via rw on the kernel command line.
+   * 
+   * Now, we want to support a read-only /sysroot: https://github.com/ostreedev/ostree/issues/1265
+   * And the way that's currently implemented is in ostree-remount.service.
+   * Ideally systemd would support "Options=bind,rw" to force on a writable bind mount,
+   * but it currently doesn't.  So we handle the ordering here.
+   */
+  if (sysroot_currently_writable)
```

Comment:
I'm still confused by this; doesn't `ostree-remount` always remount `/var` anyway? So can't we unconditionally run `Before=ostree-remount.service`?

Reply:
OK, you're right we can clean this up a lot.  Done!